Potential NPE thrown from QuarkusClassLoader.getResources

### Describe the bug

There's an issue observed on the Camel Quarkus nightly build with Quarkus `999-SNAPSHOT`.

The following change:

https://github.com/quarkusio/quarkus/commit/9daa467417b84c812ee04da5d909c79d5a729580

Causes the `camel-quarkus-kotlin-dsl` tests to fail:

```
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:233)
	at java.base/java.util.ImmutableCollections$List12.<init>(ImmutableCollections.java:563)
	at java.base/java.util.List.of(List.java:937)
	at io.quarkus.bootstrap.classloading.ClassPathElement.getResources(ClassPathElement.java:147)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.getResources(QuarkusClassLoader.java:270)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.getResources(QuarkusClassLoader.java:210)
	at kotlin.script.experimental.jvm.util.JvmClasspathUtilKt.rawClassPathFromKeyResourcePath(jvmClasspathUtil.kt:139)
	at kotlin.script.experimental.jvm.util.JvmClasspathUtilKt.classPathFromTypicalResourceUrls(jvmClasspathUtil.kt:152)
	at kotlin.script.experimental.jvm.util.JvmClasspathUtilKt$classpathFromClassloader$1.invoke(jvmClasspathUtil.kt:96)
	at kotlin.script.experimental.jvm.util.JvmClasspathUtilKt$classpathFromClassloader$1.invoke(jvmClasspathUtil.kt:77)
```

Seems that `MemoryClassPathElement` uses the default impl of [`ClassPathElement.getResources`](https://github.com/quarkusio/quarkus/blob/9d92610e65e79a44cf2b5773bfbd8f57585d1acb/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathElement.java#L146-L148). If `getResource(name)` returns `null`, then it causes an NPE because it's passed into `List.of()`.


### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

1. Clone the [camel-quarkus](https://github.com/apache/camel-quarkus) project
2. `git checkout quarkus-main`
3. `mvn clean install -Dquickly -T1C`
4. `cd integration-tests/kotlin-dsl`
5. `mvn clean test`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathElement.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathElement.java
@@ -144,6 +144,7 @@ public interface ClassPathElement extends Closeable {
     };
 
     default List<ClassPathResource> getResources(String name) {
-        return List.of(getResource(name));
+        ClassPathResource resource = getResource(name);
+        return resource == null ? List.of() : List.of(resource);
     }
 }

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/PathTreeClassPathElement.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/PathTreeClassPathElement.java
@@ -113,7 +113,7 @@ public class PathTreeClassPathElement extends AbstractClassPathElement {
         final String sanitized = sanitize(name);
         final Set<String> resources = this.resources;
         if (resources != null && !resources.contains(sanitized)) {
-            return null;
+            return List.of();
         }
         List<ClassPathResource> ret = new ArrayList<>();
         apply(tree -> {
@@ -123,7 +123,7 @@ public class PathTreeClassPathElement extends AbstractClassPathElement {
 
                 }
             });
-            return null;
+            return List.of();
         });
         return ret;
     }

